### PR TITLE
Added Buildah tests for stack package

### PR DESCRIPTION
**Pull request by CameronMcWilliam**
Adding tests for PR #839

### CODE REVIEW

File: functest/package_test.go [hunk 1]
```diff
@@ -83,8 +83,40 @@ func TestPackageDockerOptions(t *testing.T) {
...
+	// the usual core temp directory
+	sandbox.ProjectDir = filepath.Join(sandbox.ProjectDir, "starter")
+
+	args := []string{"stack", "package", "--buildah"}
+	output, err := cmdtest.RunAppsody(sandbox, args...)
+	if err != nil {
+		t.Fatal(err)
+	} else {
+		if !strings.Contains(output, "[Buildah] Writing manifest to image destination") {
+			t.Errorf("String \"[Buildah] Writing manifest to image destination\" not found in output")
```
Comment (EnriqueL8) [inline functest/package_test.go]: What exactly are we testing here? That outputs suggests that the command has passed, and the image is built?

File: functest/package_test.go [hunk 2]
```diff
@@ -83,8 +83,40 @@ func TestPackageDockerOptions(t *testing.T) {
...
+	} else {
+		t.Logf("Copied %s to %s", stackDir, sandbox.ProjectDir)
+	}
+
+	// Because the 'starter' folder has been copied, the stack.yaml file will be in the 'starter'
+	// folder within the temp directory that has been generated for sandboxing purposes, rather than
+	// the usual core temp directory
+	sandbox.ProjectDir = filepath.Join(sandbox.ProjectDir, "starter")
+
+	args := []string{"stack", "package", "--buildah"}
```
Comment (EnriqueL8) [inline functest/package_test.go]: Do we need a `--format=docker` here?

File: functest/package_test.go [hunk 3]
```diff
@@ -106,9 +138,16 @@ func TestPackageBuildahOptions(t *testing.T) {
 	sandbox.ProjectDir = filepath.Join(sandbox.ProjectDir, "starter")
 
 	args := []string{"stack", "package", "--buildah", "--buildah-options", "--format=docker"}
-	_, err = cmdtest.RunAppsody(sandbox, args...)
+	output, err := cmdtest.RunAppsody(sandbox, args...)
 	if err != nil {
 		t.Fatal(err)
+	} else {
+		if !strings.Contains(output, "format=docker") {
```
Comment (EnriqueL8) [inline functest/package_test.go]: why not include the dashes in the Contains?

File: cmd/stack_package.go [hunk 4]
```diff
@@ -272,6 +264,14 @@ The packaging process builds the stack image, generates the "tar.gz" archive fil
 				cmdArgs = append(cmdArgs, options...)
 			}
 
+			labelPairs := CreateLabelPairs(labels)
```
Comment (EnriqueL8) [inline cmd/stack_package.go]: Could this change also be an issue in `build.go` when using build options?

File: .travis.yml [hunk 5]
```diff
@@ -33,6 +33,13 @@ jobs:
     - <<: *test-template
       name: Full Test on Linux
       os: linux
+      dist: xenial
+      before_install:
```
Comment (EnriqueL8) [inline .travis.yml]: Could you put this link: https://github.com/containers/buildah/blob/master/install.md a line above to show where we have taken these commands from?

File: functest/package_test.go (hunk 2 above)
Comment (CameronMcWilliam) [reply functest/package_test.go]: There is a test with build Options and one without build options passed

File: functest/package_test.go (hunk 3 above)
Comment (CameronMcWilliam) [reply functest/package_test.go]: Don't think the dashes are passed through in the Buildah command

File: cmd/stack_package.go (hunk 4 above)
Comment (CameronMcWilliam) [reply cmd/stack_package.go]: Think in `build.go` the `buildOptions` are already before labels are added

**Pull request merged**